### PR TITLE
Bugfix if workingDirectory is missing in MergedJobDataMap

### PR DESCRIPTION
**Pull request by viebrix**
This could be one possible correction for issue:
https://github.com/quartznet/quartznet/issues/2405
But there are also other possible solutions... but also feel free to ignore this pull request

(I wasn't able to test this code, because other changes in quartz.net were not compatible with by project and dependencies)

### CODE REVIEW

File: src/Quartz.Jobs/Job/NativeJob.cs [hunk 1]
```diff
@@ -131,7 +131,11 @@ public virtual ValueTask Execute(IJobExecutionContext context)
             consumeStreams = data.GetBooleanValue(PropertyConsumeStreams);
         }
 
-        var workingDirectory = data.GetString(PropertyWorkingDirectory);
+        string? workingDirectory = null;
```
Comment (jafin) [inline src/Quartz.Jobs/Job/NativeJob.cs:134]: You should be able to shorten this to just
```c#
data.TryGetString(PropertyWorkingDirectory, out var workingDirectory);
```

**Comment by jafin** [comment]
Thanks for the fix.  This highlighted another potential issue when adding a quick test, may I suggest adding the following test code to your PR

```c#
// Tests\Job\NativeJobTests.cs
[TestFixture]
public class NativeJobTests
{
    [Test]
    public void TestNativeJob()
    {
        var job = new NativeJob();
        var context = TestUtil.NewJobExecutionContextFor(job);
        context.MergedJobDataMap.Put(NativeJob.PropertyCommand, "Test");
        //When
        Action act = () => job.Execute(context);
        act.Should().NotThrow<Exception>();
    }
}
```

and NativeJob line 122: `PropertyParameters` is also optional and suffers from the same issue you've identified, so something like:

```c#
	data.TryGetString(PropertyParameters, out var parameters);
	parameters ??= string.Empty;
```

**Comment by viebrix** [comment]
Thanks for the fast answer and your help! In meantime I have added default values for workingDirectory in my project. I didn't know that you react as fast as you did - and I needed a solution till today. Thanks a lot!

**Comment by lahma** [comment]
I'll backport to `main` manually, thanks for the 3.x fix.
